hail/src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -98,6 +112,8 @@ trait CodecSpec extends Serializable {
 
   def buildDecoder(t: PType, requestedType: PType): (InputStream) => Decoder
 
+  def buildNativeEncoder(t: PType): (OutputStream, Iterator[RegionValue]) => Unit
```

Comment:
I think this is right.  The other Specs, like BufferSpec, should also have buildNativeInputBuffer, etc.  Then you should be able to call down the entire Spec stack generating C++ and bottom out at {Input, Output}Stream.